Shopist/ShopistUITests/ShopistUITests.swift

```diff
@@ -1,18 +1,57 @@
 /*
-* Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
-* This product includes software developed at Datadog (https://www.datadoghq.com/).
-* Copyright 2019-2020 Datadog, Inc.
-*/
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2019-2020 Datadog, Inc.
+ */
 
 import XCTest
 
 class ShopistUITests: XCTestCase {
-    override func setUp() {
+    override func setUp() { // swiftlint:disable:this empty_xctest_method
         continueAfterFailure = false
     }
 
-    func testAppCanBeLaunched() {
+    func testRunFullCheckoutFlow() {
+        let sessionCount = 3
         let app = XCUIApplication()
-        app.launch()
+        for sessionIndex in 0...sessionCount {
+            app.launch()
+
+            app.collectionViews.firstMatch.goToOffScreenCell("lastCell")?.tap()
+            app.collectionViews.firstMatch.goToOffScreenCell("lastCell")?.tap()
+
+            let navBar = app.navigationBars.firstMatch
+            navBar.buttons["cart.badge.plus"].tap()
+            navBar.buttons["cart"].tap()
+
+            let cardButton = app.navigationBars["Cart"].buttons["creditcard"]
+            cardButton.tap()
+            if !app.alerts.firstMatch.waitForExistence(timeout: 5.0) {
+                fatalError("Alert should show up!")
+            }
+            app.alerts.firstMatch.buttons.firstMatch.tap()
+
+            if sessionIndex == sessionCount {
+                Thread.sleep(forTimeInterval: 5.0)
+            }
+            app.terminate()
```

Comment:
This is very hard to read as it's tightly coupled with implementation details. It will be also hard to maintain on even small UI changes and impossible to reuse. How about using the [Page Objects](https://github.com/DataDog/dd-sdk-ios/blob/rum/Tests/DatadogIntegrationTests/UITestsHelpers.swift) approach (it's kind of a standard in iOS UI Testing)? This is exactly what we use for [RUM integration tests](https://github.com/DataDog/dd-sdk-ios/blob/rum/Tests/DatadogIntegrationTests/UITestsHelpers.swift).

The PO is just a basic abstraction of the screens. With just few changes, this test will become something like this:
```swift
let app = ShopistApplication()
let categoriesListsScreen = app.launch()
let productsListScreen = categoriesListsScreen.tapTheLastCategory()
let productScreen = productsListScreen.tapTheLastProduct()
productScreen.addToCart()
let cartScreen = productScreen.openCart()
let completionAlert = cartScreen.payWithCreditCard()
app.terminate()
``` 

It's much more readable than direct calls to `XCUI` elements. When building next scenarios, we will just copy & paste the interesting parts, gaining the code reuse. And when we slightly change the UI, we won't have to modify tests scenarios each time.

Reply:
after discussing with @xgouchet, our full checkout flow scenario seems enough for demos (i just need to add some fake resources, spans and logs to pages)
i can add comments to UI tests for readability but it's not necessary to add more complexity for now (as we don't reuse them)

Reply:
> i can add comments to UI tests for readability but it's not necessary to add more complexity for now (as we don't reuse them)

Why adding comments instead of naming things in a better way 🤔? It's not a big deal to move the call to a separate class/method, no?